### ISSUE TITLE
conda forces downgrade of spyder when installing keras

### ISSUE THREAD

**Issue by tdpu** [spyder-ide/spyder#4222]
Hi, 

May not be the most appropriate place to raise this issue, but perhaps someone knows what to do?  I just recently tried to install keras from conda-forge and conda prompts me to downgrade Spyder from 3.1.3 to 2.3.8 on what seems to be an incorrect dependency conflict?

I'm not sure this has something to do with keras specifically, but any insight is appreciated.

snippet:

```
C:\Users\tdepe>conda install -c conda-forge keras
Fetching package metadata .............
Solving package specifications: .

Package plan for installation in environment C:\Users\tdepe\Anaconda3:

The following NEW packages will be INSTALLED:

    keras:     1.0.7-py35_0      conda-forge
    patch:     2.5.9-1
    theano:    0.8.2-py35_0      conda-forge

The following packages will be UPDATED:

    anaconda:  2.5.0-np110py35_0             --> custom-py35_0

The following packages will be SUPERCEDED by a higher-priority channel:

    conda:     4.3.13-py35_0                 --> 4.2.13-py35_0 conda-forge
    conda-env: 2.6.0-0                       --> 2.6.0-0       conda-forge

The following packages will be DOWNGRADED due to dependency conflicts:

    spyder:    3.1.3-py35_0                  --> 2.3.8-py35_1

Proceed ([y]/n)? y
```

## Versions and main components

* Spyder Version: 3.1.3
* Python Version: 3.5
* Qt Version: 4.8.7
* PyQt Version: 4.11.4
* Operating system: Windows 7 64-bit

## Dependencies

IPython >=1.0      :  4.0.3 (OK)
jedi >=0.8.1;<0.9.0:  0.9.0 (NOK)
matplotlib >=1.0   :  1.5.1 (OK)
pandas >=0.13.1    :  0.17.1 (OK)
pep8 >=0.6         :  1.7.0 (OK)
pyflakes >=0.6.0   :  1.0.0 (OK)
pygments >=1.6     :  2.1 (OK)
pylint >=0.25      :  1.5.4 (OK)
qtconsole >=4.0    :  4.2.1 (OK)
rope >=0.9.2       :  0.9.4-1 (OK)
sphinx >=0.6.6     :  1.3.1 (OK)
sympy >=0.7.3      :  0.7.6.1 (OK)
zmq >=2.1.11       :  15.2.0 (OK)

**Comment by ccordoba12** [comment]
I really don't why this is happening but you should create a new environment to play with conda-forge packages.

I think that should solve your problem.

**Comment by tdpu** [comment]
I believe I've solved the problem. 

The issue (for anyone who's interested) I had issued the command to install keras from conda-forge:
`conda install -c conda-forge keras`

prompting my conda channel list to take conda-forge as a higher priority channel.  This resulted in a few dependencies requiring older versions of conda and conda-env.  The install command then downgraded those packages and caused further subsequent downgrades. 

I'd also somehow got a copy of `spyder-app` and `spyder` installed and my environment was taking the conda-forge `spyder-app=2.3.8` as the highest priority.

The steps I took:
- remove the offending packages
`conda remove keras`
`conda remove spyder-app`
- add conda-forge channel as a lower priority channel
`conda config --append channels conda-forge`
- update to get the proper version of Spyder back (3.1.3)
`conda update spyder`
- install keras without superceding my original environment
`conda install keras`

Appears to be working fine now.
